src/coreclr/jit/gentree.cpp

```diff
@@ -21295,19 +21295,104 @@ GenTree* Compiler::gtNewSimdBinOpNode(
+                    (!compOpportunisticallyDependsOn(InstructionSet_AVX512) &&
+                     (simdSize > 32 || (varTypeIsInt(simdBaseType) && simdSize == 32))) ||
+                    (!compOpportunisticallyDependsOn(InstructionSet_AVX) && simdSize > 16) || simdSize == 64)
+                {
+                    var_types divType  = simdSize == 64 ? TYP_SIMD32 : TYP_SIMD16;
+                    GenTree*  op1Dup   = fgMakeMultiUse(&op1);
+                    GenTree*  op2Dup   = fgMakeMultiUse(&op2);
+                    GenTree*  op1Lower = gtNewSimdGetLowerNode(divType, op1, simdBaseJitType, simdSize);
+                    GenTree*  op2Lower = gtNewSimdGetLowerNode(divType, op2, simdBaseJitType, simdSize);
+                    GenTree*  divLower =
+                        gtNewSimdBinOpNode(GT_DIV, divType, op1Lower, op2Lower, simdBaseJitType, simdSize / 2);
+                    GenTree* op1Upper = gtNewSimdGetUpperNode(divType, op1Dup, simdBaseJitType, simdSize);
+                    GenTree* op2Upper = gtNewSimdGetUpperNode(divType, op2Dup, simdBaseJitType, simdSize);
+                    GenTree* divUpper =
+                        gtNewSimdBinOpNode(GT_DIV, divType, op1Upper, op2Upper, simdBaseJitType, simdSize / 2);
+                    GenTree* divResult = gtNewSimdWithUpperNode(type, divLower, divUpper, simdBaseJitType, simdSize);
+                    return divResult;
+                }
+
+                if (varTypeIsShort(simdBaseType) || varTypeIsByte(simdBaseType))
+                {
+                    assert(simdSize == 16);
+                    if (compOpportunisticallyDependsOn(InstructionSet_AVX512))
+                    {
+                        CorInfoType cvtBaseType =
+                            varTypeIsUnsigned(simdBaseType) ? CORINFO_TYPE_UINT : CORINFO_TYPE_INT;
+                        NamedIntrinsic widenCvtIntrinsic =
+                            varTypeIsByte(simdBaseType)
+                                ? (varTypeIsSigned(simdBaseType) ? NI_AVX512_ConvertToVector512Int32
+                                                                 : NI_AVX512_ConvertToVector512UInt32)
+                                : NI_AVX2_ConvertToVector256Int32;
+                        NamedIntrinsic narrowCvtIntrinsic =
+                            varTypeIsByte(simdBaseType)
+                                ? (varTypeIsSigned(simdBaseType) ? NI_AVX512_ConvertToVector128SByte
+                                                                 : NI_AVX512_ConvertToVector128Byte)
+                                : (varTypeIsSigned(simdBaseType) ? NI_AVX512_ConvertToVector128Int16
+                                                                 : NI_AVX512_ConvertToVector128UInt16);
+                        var_types cvtType = varTypeIsByte(simdBaseType) ? TYP_SIMD64 : TYP_SIMD32;
+                        int       cvtSize = simdSize * (varTypeIsByte(simdBaseType) ? 4 : 2);
```

Comment:
nit: This should probably just be `varTypeIsByte(simdBaseType) ? 64 : 32`. It's cheaper and doesn't risk problems for `TYP_SIMD8`/`TYP_SIMD12` nodes (although those sizes shouldn't be encountered here)

Reply:
Thanks, I've simplified the expression